spec/NewTwitchApi/Auth/AuthGuzzleClientSpec.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace spec\NewTwitchApi\Auth;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Uri;
+use NewTwitchApi\Auth\AuthGuzzleClient;
+use PhpSpec\ObjectBehavior;
+
+class AuthGuzzleClientSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(AuthGuzzleClient::class);
+        $this->shouldBeAnInstanceOf(Client::class);
```

Comment:
You're not wrong, but the way I see it is that if for some reason the code is changed and AuthGuzzleClient no longer extends Client, this will catch that. Specs are supposed to describe the code/object, right? This is me describing it. It should have the type AuthGuzzleClient (default check in generated specs) and should also be an instance of Client. I feel like this describes the class instantiation well.

Reply:
I see your point of view, but that seems like it describes implementation and not behavior, and couples your implementation to Guzzle